impeller/geometry/path.cc

```diff
@@ -11,13 +11,13 @@
 
 namespace impeller {
 
-Path::Path() {
+Path::Path() : state_(std::make_shared<State>()) {
```

Comment:
So today we have Path and PathBuilder. Both are mutable right now, but we could probably simplify this implementation a lot by making Path only internally mutable. So if we knew that only PathBuilder was adding the components, we don't need to invalidate the cache ever in path methods.

Reply:
It's invalidated in mutator methods. Are you suggesting removing the mutator methods from path completely? I'm wondering if there are plans later to have a `TrackedPath` backed by impeller path. In which case it will probably need to remain mutable.

Reply:
The mutator methods are mostly used by pathbuilder. pathbuilder, doesn't need to be constantly reseting the caches, so maybe restrict the mutation via some friend class shennanigans?

Reply:
The mutator methods are used in geometry_unittests.cc at least.

Reply:
On second look those could probably easily be replaced by builder.